### DISCUSSION TITLE
Getting crash becz multiple time pullback called (Delegate must respond to locationManager:didUpdateLocations)

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/swift-composable-architecture#877] [Q&A]
https://vimeo.com/641515257 here I have make video 

crash https://github.com/AddaMeSPB/AddaMeIOS/blob/feature/git_workflow/AddameSPM/Sources/EventView/EventsReducer.swift#L153

Delegate must respond to locationManager:didUpdateLocations: becz of many time init View

many time called event pull back 
https://github.com/AddaMeSPB/AddaMeIOS/blob/feature/git_workflow/AddameSPM/Sources/TabsView/TabsReducer.swift#L30

**Comment by mbrandonw** [comment]
When you create an `EventsEnvironment` like this in the `pullback`:

```swift
      EventsEnvironment(
        ...
        locationManager: .live,
        ...
      )
```

You are actually creating a whole _new_ location manager every time this code is executed. You can see this by looking at the source `.live`:

https://github.com/pointfreeco/composable-core-location/blob/42bb4d67b9afe88dc67ac0512f1bd98aebc7c79a/Sources/ComposableCoreLocation/Live.swift#L19-L20

The correct way to use this dependency is to have a root-level environment that holds all of your dependencies, and then when pulling back pass down just the dependencies the child feature needs.

**Reply by saroar** [reply]
@mbrandonw thanks for reply this is how I solved

```swift

public let tabsReducer = Reducer<TabsState, TabsAction, TabsEnvironment>.combine(
  eventReducer.pullback(
    state: \.event,
    action: /TabsAction.event,
    environment: { _ in EventsEnvironment.live } // here fixed my this issue
  ),
...........
extension EventsEnvironment {
  public static let live: EventsEnvironment = .init(
    pathMonitorClient: .live(queue: .main),
    locationManager: .live,
    eventClient: .live(api: .build),
    backgroundQueue: .main,
    mainQueue: .main,
    userDefaults: .live()
  )
}
```
but still I have problem it should not called multiple times(tabsReducer) when I have allow/deny location permission if I am not wrong 
going to check now hope will fixed this multiple time called root whole reducer
